zerver/actions/invites.py

```diff
@@ -376,8 +377,9 @@ def do_revoke_user_invite(prereg_user: PreregistrationUser) -> None:
     # to a "revoked" status so that we can give the invited user a better
     # error message.
     content_type = ContentType.objects.get_for_model(PreregistrationUser)
-    Confirmation.objects.filter(content_type=content_type, object_id=prereg_user.id).delete()
-    prereg_user.delete()
+    with transaction.atomic():
+        Confirmation.objects.filter(content_type=content_type, object_id=prereg_user.id).delete()
+        prereg_user.delete()
     clear_scheduled_invitation_emails(email)
```

Comment:
Oh hmm, I think `clear_scheduled_invitation_emails` should be inside the atomic block, right?

Reply:
I'll do a quick follow-up commit for this.

Reply:
Added this in b057d8ea1757dfe8e67478b0f9249c8bfa224037.